app/services/charges/validators/dynamic_service.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module Charges
+  module Validators
+    class DynamicService < Charges::Validators::BaseService
+      def valid?
+        validate_billable_metric
+
+        super
+      end
+
+      private
+
+      def validate_billable_metric
+        # Only sum aggregation is compatible with Dynamic Pricing for now
+        return if charge.billable_metric.sum_agg?
+
+        add_error(field: :billable_metric, error_code: 'invalid_billable_metric_value')
```

Comment:
For custom aggregation, the validation error is done that way on field `charge_model` with error code `invalid_aggregation_type_or_charge_model`, what do you think of using the same format?

Reply:
I copied https://github.com/getlago/lago-api/blob/fc0e89db025fec4b745763c8ae5c5749f7a4214c/app/services/charges/validators/graduated_percentage_service.rb#L32 this approach. 

The annoying part is that the `validate_charge_model` method strips the field information and moves it all to `:properties`, but having a dedicated validation class is nice. I'll try changing it to how we do it for CustomAggregation :)

Reply:
Fixed!